cmake/Modules/FindMKLDNN.cmake

```diff
@@ -20,17 +20,20 @@ IF(NOT MKLDNN_FOUND)
   SET(MKLDNN_ROOT "${PROJECT_SOURCE_DIR}/third_party/ideep/mkl-dnn")
 
   if(USE_XPU) # Build oneDNN GPU library
-    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
```

Comment:
This path is because XPU on Linux is supported with GCC only at the moment. Clang is not supported currently.

Reply:
You're right that it doesn't _currently_ support Clang, but with this condition, anyone using `CC=clang` (a very common setup) will see strange errors that do not point back to this condition, where `DNNL_LIB_NAME="dnnl.lib"`. This takes a little while to debug.

Besides, even if a dependency of PyTorch does not currently support Clang, that doesn't mean that the user shouldn't be able to at least try without the build system assuming that they're on Windows.